src/sync-controller.js

```diff
@@ -6,6 +6,12 @@ import {sumDurations} from './playlist';
 import videojs from 'video.js';
 import logger from './util/logger';
 
+const getSegmentIndex = (i, playlist, currentTime = 0) => {
+  const segments = playlist.segments;
+
+  return (playlist.endList || currentTime === 0) ? i : segments.length - (i + 1);
```

Comment:
Loop from the back of the playlist when looking for a live sync point, as we are much more likely to be closer to the end.